src/integer_mod_q/polynomial_ring_zq/sample/discrete_gauss.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains algorithms for sampling according to the uniform distribution.
+
+use crate::{
+    error::MathError,
+    integer::{PolyOverZ, Z},
+    integer_mod_q::{ModulusPolynomialRingZq, PolynomialRingZq},
+    rational::Q,
+    traits::SetCoefficient,
+    utils::sample::discrete_gauss::sample_z,
+};
+
+impl PolynomialRingZq {
+    /// Initializes a new [`PolynomialRingZq`] with degree `modulus.get_degree() - 1`
+    /// and with each entry sampled independently according to the
+    /// discrete Gaussian distribution, using [`Z::sample_discrete_gauss`].
+    ///
+    /// Parameters:
+    /// - `modulus`: specifies the [`ModulusPolynomialRingZq`] over which the
+    /// ring of polynomials modulo `modulus.get_q()` is defined
+    /// - `n`: specifies the range from which [`Z::sample_discrete_gauss`] samples
+    /// - `center`: specifies the positions of the center with peak probability
+    /// - `s`: specifies the Gaussian parameter, which is proportional
+    /// to the standard deviation `sigma * sqrt(2 * pi) = s`
+    ///
+    /// Returns a fresh [`PolynomialRingZq`] instance of length `modulus.get_degree() - 1`
+    /// with coefficients chosen independently according the discrete Gaussian distribution or
+    /// a [`MathError`] if `n <= 1` or `s <= 0`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::{PolynomialRingZq, ModulusPolynomialRingZq};
+    /// use std::str::FromStr;
+    /// let modulus = ModulusPolynomialRingZq::from_str("3  1 2 3 mod 17").unwrap();
+    ///
+    /// let sample = PolynomialRingZq::sample_discrete_gauss(&modulus, 1024, 0, 1).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`InvalidIntegerInput`](MathError::InvalidIntegerInput)
+    /// if `n <= 1` or `s <= 0`.
+    ///
+    /// # Panics ...
+    /// - if the provided [`ModulusPolynomialRingZq`] has degree 0 or smaller.
+    pub fn sample_discrete_gauss(
+        modulus: &ModulusPolynomialRingZq,
+        n: impl Into<Z>,
+        center: impl Into<Q>,
+        s: impl Into<Q>,
+    ) -> Result<Self, MathError> {
+        assert!(
+            modulus.get_degree() > 0,
+            "ModulusPolynomial of degree 0 is insufficient to sample over."
+        );
+        let n = n.into();
+        let center = center.into();
+        let s = s.into();
+        let mut poly = PolynomialRingZq::from((&PolyOverZ::default(), modulus));
+
+        for index in 0..modulus.get_degree() {
+            let sample = sample_z(&n, &center, &s)?;
+            poly.set_coeff(index, &sample)?;
+        }
```

Comment:
alternatively, and better in my opinion, implement PolyOverZ::sample_discrete_gauss where you provide a maximal degree and then only call this function and wrap it, because the other one is needed anyways

or at least I could use that too :D